app/src/main/java/io/nekohasekai/sagernet/fmt/ConfigBuilder.kt

```diff
@@ -412,10 +412,19 @@ fun buildConfig(
                         }
 
                         val useBrutal = bean.serverBrutal && bean.canBrutal()
-                        if (
-                            (!muxApplied && proxyEntity.needCoreMux()) ||
-                            useBrutal
-                        ) {
+                        var useMux = (!muxApplied && proxyEntity.needCoreMux()) || useBrutal
+                        try {
+                            val outboundMuxState = bean.javaClass.getField("muxState").get(bean)
+                            if (outboundMuxState is Int) {
+                                // prioritize profile mux state to overall setting
+                                when(outboundMuxState) {
```

Comment:
In the new commit, I changed the behavior of the UI so that when mux is set to Off, brutal will be turned off and cannot be turned on until mux is either On or Default (brutal is practically disabled), this way their interaction is more transparent and the case where mux is Off and brutal is on will never occur. I also added constants for muxState and used them in the code.